### PR TITLE
added test case for iloc function if it returns the same output for b…

### LINKED ISSUE #40933: BUG: Inconsistency for iloc between np.array and pd.array
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

We got some inconsistencies with iloc between setting a numpy array and a pandas array

```python
df = pd.DataFrame(data={
    'col1': [1, 2, 3, 4],
    'col2': [3, 4, 5, 6],
    'col3': [6, 7, 8, 9],
})
rhs = pd.array([1, 2, 3])
df.iloc[[1, 2, 3]] = rhs

```

#### Problem description
This returns

```
   col1  col2  col3
0     1     3     6
1     1     1     1
2     2     2     2
3     3     3     3
```

When switching the rhs to a numpy array, this returns

```
rhs = np.array([1, 2, 3])

   col1  col2  col3
0     1     3     6
1     1     2     3
2     1     2     3
3     1     2     3
```

Same happens when using ``[[1, 2, 3], :]`` as indexer

#### Expected Output

We should be consistent here, if we want to change iloc for the numpy case I think we need to deprecate?

Is this already known?
cc @jbrockmendel @jorisvandenbossche 

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : c8e077fa87a1e60bd6f8b7a9e5d718092bf70739
python           : 3.8.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.8.0-48-generic
Version          : #54~20.04.1-Ubuntu SMP Sat Mar 20 13:40:25 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.0.dev0+1320.gc8e077fa87.dirty
numpy            : 1.20.0
pytz             : 2020.5
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 49.6.0.post20210108
Cython           : 0.29.21
pytest           : 6.2.1
hypothesis       : 6.0.2
sphinx           : 3.4.3
blosc            : None
feather          : None
xlsxwriter       : 1.3.7
lxml.etree       : 4.6.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.8.5
fastparquet      : 0.5.0
gcsfs            : 0.7.1
matplotlib       : 3.3.3
numexpr          : 2.7.2
odfpy            : None
openpyxl         : 3.0.6
pandas_gbq       : None
pyarrow          : 1.0.0
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.6.0
sqlalchemy       : 1.3.22
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.16.2
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.52.0
None

</details>

### PR DESCRIPTION
**Pull request by parkyo**
…oth numpy and pandas arrays to fix the issue #40933 

- [x] closes #40933 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_iloc.py [hunk 1]
```diff
@@ -73,6 +74,24 @@ def test_iloc_getitem_multiple_items():
...
+    # test if iloc returns the same output for numpy array input and integer array input
+    df = DataFrame(
+        data={
+            "col1": [1, 2, 3, 4],
+            "col2": [3, 4, 5, 6],
+            "col3": [6, 7, 8, 9],
+        }
+    )
+    df_np = df
+    df_pd = df
```
Comment (phofl) [inline pandas/tests/indexing/multiindex/test_iloc.py:87]: See my comment at the issue, this does not test the behavior correctly

Comment (parkyo) [reply pandas/tests/indexing/multiindex/test_iloc.py]: I see what is wrong in my tests. I rewrote the test and fixed the error. Please confirm!

**Comment by phofl** [comment]
We can not cast to numpy here, our ExtensionArrays behave different

**Comment by simonjayhawkins** [comment]
closing as stale. ping when ready to continue and i'll reopen.

**Pull request closed without merging**